dev/merge_arrow_pr.py

```diff
@@ -418,11 +418,23 @@ def merge_pr(self, number, commit_title, commit_message):
         }
         response = requests.put(url, headers=self.headers, json=payload)
         result = response.json()
-        if response.status_code != 200 and 'merged' not in result:
+        if response.status_code == 200 and 'merged' in result:
```

Comment:
btw based on the API if the result is 200 the merge was successful and `merged` will always be true.
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#merge-a-pull-request
I am sure we can mark this as `False` if those are not successful.